File: app/src/main/java/com/example/keybindhelper/dao/DateConverter.java
package com.example.keybindhelper.dao;

import androidx.room.TypeConverter;
import java.util.*;

public class DateConverter {
    @TypeConverter
    public static Date toDate(Long timestamp){
        return (timestamp == null)? null : new Date(timestamp);
    }
    @TypeConverter
    public static Long toTimestamp(Date date){
        return date.getTime();
    }
}

File: app/src/main/java/com/example/keybindhelper/dao/DateConverter.kt
package com.example.keybindhelper.dao

import androidx.room.TypeConverter
import java.util.*

class DateConverter {
    companion object    {
        @JvmStatic
        @TypeConverter
        fun toDate(timestamp: Long?): Date? {
            return if (timestamp == null) null else Date(timestamp)
        }

        @JvmStatic
        @TypeConverter
        fun toTimestamp(date: Date?): Long? {
            return date?.time
        }
    }

}